addons/product/models/res_partner.py

```diff
@@ -30,9 +30,17 @@ def _inverse_product_pricelist(self):
         )
         default_for_country = pls and pls[0]
         actual = self.env['ir.property'].get('property_product_pricelist', 'res.partner', 'res.partner,%s' % self.id)
-
+        # We want to keep the current pricelist if it's still valid for the new country
+        # or if the country would be the same, like in the case of an state change that
+        # triggers this method and could lead to lose the original pricelist
+        update_country = False
+        if default_for_country and actual and default_for_country.id != actual.id:
+            if not actual.country_group_ids:
+                update_country = default_for_country.currency_id == actual.currency_id
```

Comment:
Hello @chienandalu 

I'm not ok with this change.
According to me, it should be more logical to change the pricelist if the actual pricelist is in a different currency than the default_for_country
So to write instead:
```
update_country = default_for_country.currency_id != actual.currency_id
```
Let's consider two pricelists P1 and P2 such as
P1 has no country group and is in $
P2 has the country group Benelux (Belgium, Netherland, Luxembourg) and is in €
Let's consider a partner P with US as country and with P1 as pricelist.
If you change the country of P1 to Belgium, we expect that his pricelist will be updated to P2 because it has no sens to keep his pricelist in $ if he is from Belgium
But with your change it will keep P1

Reply:
Sure. You're right. That was my intention, thanks :)